### ISSUE TITLE
Doesn't compile with vector-0.13 (with --allow-newer in order to ignore the bound from .cabal)

### ISSUE THREAD

**Issue by Mikolaj** [haskellari/strict-containers#6]
Compiles fine with vector-0.12.3.1. The failure is 

```
[41 of 44] Compiling Data.Strict.Vector.Autogen.Mutable ( src/Data/Strict/Vector/Autogen/Mutable.hs, dist/build/Data/Strict/Vector/Autogen/Mutable.o, dist/build/Data/Strict/Vector/Autogen/Mutable.dyn_o )

src/Data/Strict/Vector/Autogen/Mutable.hs:161:3: error:
    Data constructor not in scope:
      INTERNAL_CHECK :: t2 -> String -> String -> Bool -> m2 () -> m ()
    |
161 |   INTERNAL_CHECK(check) "moveBackwards" "not a backwards move" (dstOff < srcOff)
    |   ^^^^^^^^^^^^^^

src/Data/Strict/Vector/Autogen/Mutable.hs:161:18: error:
    Variable not in scope: check
    |
161 |   INTERNAL_CHECK(check) "moveBackwards" "not a backwards move" (dstOff < srcOff)
    |                  ^^^^^

src/Data/Strict/Vector/Autogen/Mutable.hs:168:3: error:
    Data constructor not in scope:
      INTERNAL_CHECK :: t1 -> String -> String -> Bool -> m1 () -> m ()
    |
168 |   INTERNAL_CHECK(check) "moveForwardsSmallOverlap" "not a forward move" (dstOff > srcOff)
    |   ^^^^^^^^^^^^^^

src/Data/Strict/Vector/Autogen/Mutable.hs:168:18: error:
    Variable not in scope: check
    |
168 |   INTERNAL_CHECK(check) "moveForwardsSmallOverlap" "not a forward move" (dstOff > srcOff)
    |                  ^^^^^

src/Data/Strict/Vector/Autogen/Mutable.hs:179:3: error:
    Data constructor not in scope:
      INTERNAL_CHECK :: t0 -> String -> String -> Bool -> m0 () -> m ()
    |
179 |   INTERNAL_CHECK(check) "moveForwardsLargeOverlap" "not a forward move" (dstOff > srcOff)
    |   ^^^^^^^^^^^^^^

src/Data/Strict/Vector/Autogen/Mutable.hs:179:18: error:
    Variable not in scope: check
    |
179 |   INTERNAL_CHECK(check) "moveForwardsLargeOverlap" "not a forward move" (dstOff > srcOff)
    |                  ^^^^^
Error: cabal: Failed to build strict-containers-0.1 (which is required by
test:shortTestForCI from horde-ad-0.1.0.0, test:minimalTest from
horde-ad-0.1.0.0 and others). See the build log above for details.
```

**Comment by sjakobi** [comment]
`strict-containers-serialise` and `strict-containers-tests` are blocked on https://github.com/well-typed/cborg/issues/302.

**Comment by infinity0** [comment]
This issue is blocked on well-typed/cborg#302 but it's not the only thing it's blocked on.

This package is autogenerated from vector-0.12.3.0 (details in docs) so in general it's not expected to work with newer versions. The bound from .cabal is very very hard. See `strict-containers/regen.sh` for details.

**Comment by infinity0** [comment]
Tentative PR in #10 but I'm having trouble debugging the CI failure, help would be appreciated.

**Comment by Mikolaj** [comment]
Hey, that's great news! Anything else holding up the release?

**Comment by infinity0** [comment]
Whoops I forgot, will do that shortly.

**Comment by infinity0** [comment]
https://hackage.haskell.org/package/strict-containers

**Comment by Mikolaj** [comment]
Thank you. I've tested the new release in https://github.com/Mikolaj/horde-ad and it works great.

**Comment by infinity0** [comment]
Nice, glad to see it's being used! :+1:
